cognite/client/data_classes/vision.py

```diff
@@ -0,0 +1,254 @@
+    "people_predictions": "images.ObjectDetection",
+    "personal_protective_equipment_predictions": "images.ObjectDetection",
+}
+
+
+@dataclass
+class ThresholdParameter:
+    threshold: Optional[float] = None
+
+
+@dataclass
+class AssetTagDetectionParameters(VisionResource, ThresholdParameter):
+    partial_match: Optional[bool] = None
+    asset_subtree_ids: Optional[List[int]] = None
+
+
+@dataclass
+class TextDetectionParameters(VisionResource, ThresholdParameter):
+    pass
+
+
+@dataclass
+class PeopleDetectionParameters(VisionResource, ThresholdParameter):
+    pass
+
+
+@dataclass
+class IndustrialObjectDetectionParameters(VisionResource, ThresholdParameter):
+    pass
+
+
+@dataclass
+class PersonalProtectiveEquipmentDetectionParameters(VisionResource, ThresholdParameter):
+    pass
+
+
+@dataclass
+class FeatureParameters(VisionResource):
+    text_detection_parameters: Optional[TextDetectionParameters] = None
+    asset_tag_detection_parameters: Optional[AssetTagDetectionParameters] = None
+    people_detection_parameters: Optional[PeopleDetectionParameters] = None
+    industrial_object_detection_parameters: Optional[IndustrialObjectDetectionParameters] = None
+    personal_protective_equipment_detection_parameters: Optional[PersonalProtectiveEquipmentDetectionParameters] = None
+
+
+class VisionJob(ContextualizationJob):
+    def update_status(self) -> str:
+        # Override update_status since we need to handle the vision-specific
+        # edge case where we also record failed items per batch
+        data = (
+            self._cognite_client.__getattribute__(self._JOB_TYPE.value)._get(f"{self._status_path}{self.job_id}").json()
+        )
+        self.status = data["status"]
+        self.status_time = data.get("statusTime")
+        self.start_time = data.get("startTime")
+        self.created_time = self.created_time or data.get("createdTime")
+        self.error_message = data.get("errorMessage") or data.get("failedItems")
+        self._result = {k: v for k, v in data.items() if k not in self._COMMON_FIELDS}
+        assert self.status is not None
+        return self.status
+
+
+class VisionExtractItem(CogniteResource):
+    def __init__(
+        self,
+        file_id: int = None,
+        predictions: Dict[str, Any] = None,
+        file_external_id: str = None,
+        error_message: str = None,
+        cognite_client: "CogniteClient" = None,  # noqa: F821
+    ) -> None:
+        """Data class for storing predictions for a single image file"""
+        self.file_id = file_id
+        self.file_external_id = file_external_id
+        self.error_message = error_message
+        self.predictions = self._process_predictions_dict(predictions) if isinstance(predictions, Dict) else predictions
+
+        self._predictions_dict = predictions  # The "raw" predictions dict returned by the endpoint
+        self._cognite_client = cast("CogniteClient", cognite_client)  # noqa: F821
+
+    @classmethod
+    def _load(
```

Comment:
Suggestions on how to fix this immediatly?

Reply:
[4c6942f](https://github.com/cognitedata/cognite-sdk-python/pull/1014/commits/4c6942fe2808a3189c7fa3de8d283a9dbdbb5f3c)